numpyro/infer/mcmc.py

```diff
@@ -1084,7 +1077,9 @@ def _single_chain_mcmc(self, rng_key, init_state, init_params, args, kwargs, col
             init_state = self.sampler.init(rng_key, self.num_warmup, init_params,
                                            model_args=args, model_kwargs=kwargs)
         if self.postprocess_fn is None:
-            self.postprocess_fn = self.sampler.postprocess_fn(args, kwargs)
+            postprocess_fn = self.sampler.postprocess_fn(args, kwargs)
+        else:
+            postprocess_fn = self.postprocess_fn
```

Comment:
I believe this is the intended behavior: if users provide postprocess_fn, we use it; otherwise, use `sampler.postprocess_fn` with "current" args, kwargs. I am not sure why this does not raise any issue previously.

Reply:
The issue is when using args, kwargs at later runs, when args, kwargs change. Before this pr, I think we use the same args, kwargs for different runs here.